0-module-and-module-source.emu

```diff
@@ -0,0 +1,614 @@
+          </td>
+        </tr>
+        <tr>
+            <td>
+              [[HasTLA]]
+            </td>
+            <td>
+              a Boolean
+            </td>
+            <td>
+              Whether this module source is individually asynchronous. Having an asynchronous dependency does not mean this field is *true*.
+            </td>
+        </tr>
+        <tr>
+            <td>
+              [[RequestedModules]]
+            </td>
+            <td>
+              a List of Strings
+            </td>
+            <td>
+              A List of all the |ModuleSpecifier| strings used by the module source represented by this record to request the importation of a module. The List is source text occurrence ordered.
+            </td>
+        </tr>
+        <tr>
+            <td>
+              [[HostDefined]]
+            </td>
+            <td>
+              anything (default value is *undefined*)
+            </td>
+            <td>
+              Field reserved for use by host environments that need to associate additional information with a module source.
+            </td>
+          </tr>
+      </table>
+    </emu-table>
+</emu-clause>
+
+<emu-clause id="sec-import-calls">
+    <h1>Import Calls</h1>
+
+    <emu-clause id="sec-import-call-runtime-semantics-evaluation" type="sdo">
+      <h1>Runtime Semantics: Evaluation</h1>
+
+      <emu-grammar>ImportCall : `import` `(` AssignmentExpression `)`</emu-grammar>
+      <emu-alg>
+        1. Let _referencingScriptOrModule_ be GetActiveScriptOrModule().
+        1. Let _argRef_ be the result of evaluating |AssignmentExpression|.
+        1. <del>Let _specifier_ be ? GetValue(_argRef_).</del>
+        1. <ins>Let _specifierOrModule_ be ? GetValue(_argRef_).</ins>
+        1. Let _promiseCapability_ be ! NewPromiseCapability(%Promise%).
+        1. <ins>If Type(_specifierOrModule_) is Object that has a [[ModuleSourceInstance]] internal slot, then</ins>
+          1. <ins>Let _moduleRecord_ be _specifierOrModule_.[[Module]].</ins>
+          1. <ins>Perform HostImportModuleRecordDynamically(_moduleRecord_, _promiseCapability_).</ins>
+        1. <ins>Else,</ins>
+          1. Let _specifierString_ be Completion(ToString(_specifierOrModule_)).
+          1. IfAbruptRejectPromise(_specifierString_, _promiseCapability_).
+          1. Perform HostImportModuleDynamically(_referencingScriptOrModule_, _specifierString_, _promiseCapability_).
+        1. Return _promiseCapability_.[[Promise]].
+      </emu-alg>
+    </emu-clause>
+</emu-clause>
+
+<emu-clause id="sec-modulesource-objects">
+    <h1>ModuleSource Objects</h1>
+    <emu-clause id="sec-modulesource-abstracts">
+        <h1>ModuleSource Abstract Operations</h1>
+
+        <emu-clause id="sec-parsemodulesource" type="abstract operation">
```

Comment:
This algr looks very similar to [ParseModule](https://tc39.es/ecma262/#sec-parsemodule). Can you use `<ins>` and `<del>` to indicate the changes?

Reply:
I have added a note below this operation related to this. I don't it will be a modifcation, but a refactor that splits ParseModule into two operations.